### PR TITLE
docs : add Hardware Security Recommendation

### PR DESCRIPTION
**Pull request by shjala**
Add the initial version of HW security recommendation document.

### CODE REVIEW

File: docs/SECURITY-HW-REC.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Firmware
+
+Make sure the firmware's authenticity and integrity are checked before execution and during update. Update the firmwares to the latest available version and make sure the implementation has mechanisms like rollback protection in place to prevent flashing an older vulnerable version to the system. Updating firmwares via [UEFI capsules](https://github.com/fwupd/fwupd) it is considered and will be supported by EVE.
+
+## Intel Hyperthreading
+
+It is recommended to disable Hyperthreading if the system is handling sensitive information. This technology has a reputation for being flawed and paving the road for side-channel attacks and information disclosures.
```
Comment (eriknordmark) [inline docs/SECURITY-HW-REC.md:25]: Makes sense to check whether existing users of EVE does this or whether they depend on the additional performance one might see with HT. Ping me for a pointer.

**Review by eriknordmark — COMMENTED** [review]
Looks good. 
Makes sense to add a pointer from the HARDWARE-BRINGUP.md and SECURIYY.md documents to this document.

**Review by eriknordmark — COMMENTED** [review]
yetus (under checks) reports:
<div class="js-updatable-content js-socket-channel" data-channel="eyJjIjoiY2hlY2tfc3VpdGVzOjExMjI1MTg5NTcyIiwidCI6MTY3NzUyNzQ1MH0=--aa3d42d7717f6ef9cb652549ef881af460bc08a7701a5d1929bc75031aea60e6" data-url="/lf-edge/eve/actions/runs/4282209608/annotations_partial" style="box-sizing: border-box; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="actions-fullwidth-module color-bg-default color-shadow-small mb-3 pt-3 pb-1 border rounded" style="box-sizing: border-box; border: var(--primer-borderWidth-thin, 1px) solid var(--color-border-default)  !important; border-radius: var(--primer-borderRadius-medium, 6px)  !important; box-shadow: var(--color-shadow-small)  !important; background-color: var(--color-canvas-default)  !important; margin-bottom: var(--base-size-16, 16px)  !important; padding-bottom: var(--base-size-4, 4px)  !important; padding-top: var(--base-size-16, 16px)  !important;">

yetus: docs/SECURITY-HW-REC.md#L31markdownlint:MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
--


</div></div><div id="artifacts" class="js-updatable-content js-socket-channel" data-channel="eyJjIjoid29ya2Zsb3dfcnVuOjQyODIyMDk2MDg6YXJ0aWZhY3RzIiwidCI6MTY3NzUyNzQ1MH0=--20f6e13b3105a036957f5db49ae80ccd1d405229e6b8ce6eedd06e88c75a84da" data-url="/lf-edge/eve/actions/runs/4282209608/artifacts_partial" style="box-sizing: border-box; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="actions-fullwidth-module color-bg-default Box color-shadow-small mb-3 pt-3 pb-1 border rounded" style="box-sizing: border-box; border-color: var(--color-border-default); border-style: solid; border-width: var(--primer-borderWidth-thin, 1px); border-image-source:  !important; border-image-slice:  !important; border-image-width:  !important; border-image-outset:  !important; border-image-repeat:  !important; border-radius: var(--primer-borderRadius-medium, 6px); box-shadow: var(--color-shadow-small)  !important; background-color: var(--color-canvas-default); margin-bottom: var(--base-size-16, 16px)  !important; padding-bottom: var(--base-size-4, 4px)  !important; padding-top: var(--base-size-16, 16px)  !important;"><div class="mt-1 mb-3 px-3" style="box-sizing: border-box; margin-top: var(--base-size-4, 4px)  !important; margin-bottom: var(--base-size-16, 16px)  !important; padding-right: var(--base-size-16, 16px)  !important; padding-left: var(--base-size-16, 16px)  !important;"><div class="mx-0 mx-md-1" style="box-sizing: border-box; margin-right: var(--base-size-4, 4px)  !important; margin-left: var(--base-size-4, 4px)  !important;"><br class="Apple-interchange-newline">[yetus: docs/SECURITY-HW-REC.md#L31](https://github.com/lf-edge/eve/pull/3066/files#annotation_9239390264)
markdownlint:MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]

That is supposed to also appear as annotations under the diffs i.e., in https://github.com/lf-edge/eve/pull/3066/files, but I don't see it there for this PR.

**Comment by shjala** [comment]
@eriknordmark please review.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by eriknordmark — APPROVED** [review]
Looks good.

**Pull request merged**